subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultTaskInputs.java

```diff
@@ -147,8 +147,10 @@ public FileCollection getSourceFiles() {
     }
 
     public Map<String, Object> getProperties() {
+        DeprecationLogger.nagUserOfDiscontinuedMethod("TaskInputs.getProperties()", "Use the property() and properties() methods to declare input properties instead.");
         GetInputPropertiesVisitor visitor = new GetInputPropertiesVisitor(task.getName());
         TaskPropertyUtils.visitProperties(propertyWalker, task, visitor);
+        //noinspection ConstantConditions
```

Comment:
What is the `noinspection` for?

Reply:
The `Factory` is `@Nullable` but of course this won't return `null` ever.